service/src/main/java/org/exoplatform/extension/exchange/service/util/CalendarConverterUtils.java

```diff
@@ -184,7 +185,8 @@ public static void convertExchangeToExoMasterRecurringCalendarEvent(CalendarEven
           }
         }
         event.setRepeatByDay(daysList.toArray(new String[0]));
-        Utils.adaptRepeatRule(event, DEFAULT_TIME_ZONE , CalendarService.PERSISTED_TIMEZONE);
+        CalendarSetting calendarSetting = getCalendarService().getCalendarSetting(username);
+        Utils.adaptRepeatRule(event, DateUtils.getTimeZone(calendarSetting.getTimeZone()), CalendarService.PERSISTED_TIMEZONE);

```

Comment:
You should add this instruction in a dedicated static method:

`public static TimeZone getUserTimeZone(String username)`

The method have to not return null in any case.

Reply:
Fixed.